From 310dcfc15495f1876f869fe64066211b84bf86a8 Mon Sep 17 00:00:00 2001
From: Sebastien Guiriec <s-guiriec@ti.com>
Date: Tue, 11 Jan 2011 08:31:40 -0600
Subject: ASoC: ABE DSP: Fix mismatch of enable/disable gainsi for mixers

Due to a bad merge from L24x branch, enable/disable ABE gains
for MIXDL1, MIXAUDUL and MIXSDT were incorrectly set.

Change-Id: Iefefc587a18b832cc1644a571025ad14e66c5690

Signed-off-by: Sebastien Guiriec <s-guiriec@ti.com>
Signed-off-by: Dan Murphy <dmurphy@ti.com>
---
 sound/soc/omap/omap-abe-dsp.c |   12 ++++++------
 1 files changed, 6 insertions(+), 6 deletions(-)

Index: Kernel/sound/soc/omap/omap-abe-dsp.c
===================================================================
--- Kernel.orig/sound/soc/omap/omap-abe-dsp.c	2011-01-19 00:19:31.413322620 +0530
+++ Kernel/sound/soc/omap/omap-abe-dsp.c	2011-01-19 00:19:37.652073914 +0530
@@ -374,9 +374,11 @@
 	if (ucontrol->value.integer.value[0]) {
 		abe->dapm[mc->shift] = ucontrol->value.integer.value[0];
 		snd_soc_dapm_mixer_update_power(widget, kcontrol, 1);
+		abe_enable_gain(MIXDL1, mc->reg);
 	} else {
 		abe->dapm[mc->shift] = ucontrol->value.integer.value[0];
 		snd_soc_dapm_mixer_update_power(widget, kcontrol, 0);
+		abe_disable_gain(MIXDL1, mc->reg);
 	}
 	pm_runtime_put_sync(&pdev->dev);
 
@@ -424,11 +426,11 @@
 	if (ucontrol->value.integer.value[0]) {
 		abe->dapm[mc->shift] = ucontrol->value.integer.value[0];
 		snd_soc_dapm_mixer_update_power(widget, kcontrol, 1);
-		abe_enable_gain(MIXDL1, mc->reg);
+		abe_enable_gain(MIXAUDUL, mc->reg);
 	} else {
 		abe->dapm[mc->shift] = ucontrol->value.integer.value[0];
 		snd_soc_dapm_mixer_update_power(widget, kcontrol, 0);
-		abe_disable_gain(MIXDL1, mc->reg);
+		abe_disable_gain(MIXAUDUL, mc->reg);
 	}
 	pm_runtime_put_sync(&pdev->dev);
 
@@ -476,11 +478,11 @@
 	if (ucontrol->value.integer.value[0]) {
 		abe->dapm[mc->shift] = ucontrol->value.integer.value[0];
 		snd_soc_dapm_mixer_update_power(widget, kcontrol, 1);
-		abe_enable_gain(MIXAUDUL, mc->reg);
+		abe_enable_gain(MIXSDT, mc->reg);
 	} else {
 		abe->dapm[mc->shift] = ucontrol->value.integer.value[0];
 		snd_soc_dapm_mixer_update_power(widget, kcontrol, 0);
-		abe_disable_gain(MIXAUDUL, mc->reg);
+		abe_disable_gain(MIXSDT, mc->reg);
 	}
 	pm_runtime_put_sync(&pdev->dev);
 
@@ -576,11 +578,9 @@
 	if (ucontrol->value.integer.value[0]) {
 		abe->dapm[mc->shift] = ucontrol->value.integer.value[0];
 		snd_soc_dapm_mixer_update_power(widget, kcontrol, 1);
-		abe_enable_gain(MIXSDT, mc->reg);
 	} else {
 		abe->dapm[mc->shift] = ucontrol->value.integer.value[0];
 		snd_soc_dapm_mixer_update_power(widget, kcontrol, 0);
-		abe_disable_gain(MIXSDT, mc->reg);
 	}
 	pm_runtime_put_sync(&pdev->dev);
 
